### PR TITLE
feat: Update to Angular v18

### LINKED ISSUE #251: Please add support for Angular v18
Angular 18 has been released, please consider adding support for Angular 18

### PR DESCRIPTION
**Pull request by moh4sa**
Upgrade the packages to the latest version of Angular. close #251

### CODE REVIEW

**Review by henrylangenhoven — APPROVED** [review]

**Comment by moh4sa** [comment]
@Cordobo  for you review

**Comment by Arthi93** [comment]
Any progress on this? Will really appreciate it for my ng 18 project :)

File: README.md [hunk 1]
```diff
@@ -227,4 +234,4 @@ Support the development of angularx-qrcode (or even see your logo here?), consid
 
 MIT License
 
-Copyright (c) 2018 - present [Andreas Jacob (Cordobo.com)](http://cordobo.com/)
+Copyright (c) 2024 - present [Andreas Jacob (Cordobo.com)](http://cordobo.com/)
```
Comment (Cordobo) [inline README.md:237]: 2018 - present is fine

File: projects/angularx-qrcode/README.md [hunk 2]
```diff
@@ -227,4 +234,4 @@ Support the development of angularx-qrcode (or even see your logo here?), consid
 
 MIT License
 
-Copyright (c) 2018 - present [Andreas Jacob (Cordobo.com)](http://cordobo.com/)
+Copyright (c) 2024 - present [Andreas Jacob (Cordobo.com)](http://cordobo.com/)
```
Comment (Cordobo) [inline projects/angularx-qrcode/README.md:237]: 2018 - present is fine

File: projects/demo-app/src/app/app.component.html [hunk 3]
```diff
@@ -354,7 +354,7 @@ <h4>Accessibility/a11y attributes (optional)</h4>
 </div>
 
 <footer id="footer">
-  <span>MIT License | Copyright © 2018 - present</span>
+  <span>MIT License | Copyright © 2024 - present</span>
```
Comment (Cordobo) [inline projects/demo-app/src/app/app.component.html:357]: 2018 - present is fine

File: package.json [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
   "name": "angularx-qrcode",
-  "version": "17.0.1",
+  "version": "18.0.0",
   "main": "dist/angularx-qrcode/bundles/angularx-qrcode.umd.js",
   "author": {
     "name": "Andreas Jacob <andreas@cordobo.de>",
     "url": "http://cordobo.com/"
   },
-  "description": "Simple QRCode generator library for Angular4/5/6/7/8/9/10/11/12/13/14/15/16/17+ and Ionic 3/4/5 using node-qrcode",
+  "description": "Simple QRCode generator library for Angular4/5/6/7/8/9/10/11/12/13/14/15/16/17/18+ and Ionic 3/4/5 using node-qrcode",
```
Comment (justvamp) [inline package.json:10]: Ionic 3/4/5/6/7/8

File: README.md (hunk 1 above)
Comment (moh4sa) [reply README.md]: reverted

File: projects/angularx-qrcode/README.md (hunk 2 above)
Comment (moh4sa) [reply projects/angularx-qrcode/README.md]: reverted

File: projects/demo-app/src/app/app.component.html (hunk 3 above)
Comment (moh4sa) [reply projects/demo-app/src/app/app.component.html]: reverted

File: package.json (hunk 4 above)
Comment (moh4sa) [reply package.json]: added

**Review by justvamp — APPROVED** [review]

**Comment by kyjus25** [comment]
What's poppin', don't mind me just watchin 👀 😆

**Comment by lvetskov** [comment]
Really looking forward to this upgrade

**Comment by moh4sa** [comment]
@Cordobo this is pending your review

**Review by Sekators — COMMENTED** [review]
v18

**Comment by siddharthcn21** [comment]
Any progress on this? really appreciate it if can be fast track this. Thanks.

**Comment by JustroX** [comment]
Just in case you need to urgently update, you may want temporarily add this to package.json while waiting 
image

**Review by NijboerLars — APPROVED** [review]

**Comment by xanather** [comment]
Angular 18 is two months old now, rather not force a dependency override. Please merge this :)

**Comment by behdi** [comment]
@Cordobo We'd really appreciate it if you could merge this sooner. This is a great library and it's the only one holding us back from updating to v18.

**Comment by A-ahmadisalem** [comment]
@Cordobo Do you have an estimated timeline for when this will be merged? This is the only package we are waiting on to complete our migration to Angular 18.

**Comment by xanather** [comment]
I think its time to decide:
1. Maintainer adds more other maintainers with write access
2. Fork this
3. Use a different package

**Comment by caboodal** [comment]
Are there any other packages?

**Comment by moh4sa** [comment]
@everyone I think this would be a good replacement for anyone who still waiting for this PR like me https://github.com/mnahkies/ng-qrcode#readme

**Comment by maxmayr95** [comment]
A merge would help us too!

**Comment by caboodal** [comment]
I'm going to look at switching to the other component tomorrow. I'll report back my findings.

**Comment by maxmayr95** [comment]
@caboodal why not just use the fix? The one with overriding angular core version? Works good for me. Tested today

**Comment by caboodal** [comment]
@maxmayr95 we are using the fix but I don't want a dependency in our apps that's not being maintained. Angular versions are coming thick and fast the next one we might not be so lucky.

**Review by Cordobo — APPROVED** [review]

**Pull request merged**